src/Facets/RelayFacet.sol

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { LibDiamond } from "../Libraries/LibDiamond.sol";
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+import { LibUtil } from "../Libraries/LibUtil.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { SwapperV2 } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+import { ECDSA } from "solady/utils/ECDSA.sol";
+
+/// @title Relay Facet
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality for bridging through Relay Protocol
+/// @custom:version 1.0.0
+contract RelayFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    /// Storage ///
+
+    // Receiver for native transfers
+    address public immutable relayReceiver;
+    address public immutable relaySolver;
+
+    /// Types ///
+
+    /// @dev Optional bridge specific struct
+    /// @param exampleParam Example parameter
+    struct RelayData {
+        bytes32 requestId;
+        address receivingAssetId;
+        bytes callData;
+        bytes signature;
+    }
+
+    /// Errors ///
+
+    error InvalidQuote();
+
+    /// Modifiers ///
+
+    modifier onlyValidQuote(
+        ILiFi.BridgeData memory _bridgeData,
+        RelayData calldata _relayData
+    ) {
+        // Verify that the bridging quote has been signed by the Relay solver
+        // as attested using the attestaion API
+        // API URL: https://api.relay.link/requests/{requestId}/signature/v2
+        bytes32 message = keccak256(
+            abi.encodePacked(
+                "\x19Ethereum Signed Message:\n32",
+                keccak256(
+                    abi.encodePacked(
+                        _relayData.requestId,
+                        block.chainid,
+                        bytes32(uint256(uint160(address(this)))),
+                        bytes32(uint256(uint160(_bridgeData.sendingAssetId))),
+                        _bridgeData.destinationChainId,
+                        bytes32(uint256(uint160(_bridgeData.receiver))),
```

Comment:
Please also add support for non-evm addresses. In the same way we handle it for DeBridge.